src/internal/pfsdb/commits.go

```diff
@@ -0,0 +1,956 @@
+type ErrChildCommitNotFound struct {
+	Repo           string
+	ParentRowID    CommitID
+	ParentCommitID string
+}
+
+func IsChildCommitNotFound(err error) bool {
+	return dbutil.IsNotNullViolation(err, "to_id")
+}
+
+// Error satisfies the error interface.
+func (err ErrChildCommitNotFound) Error() string {
+	if id := err.ParentRowID; id != 0 {
+		return fmt.Sprintf("parent commit of commit id=%d not found", id)
+	}
+	return fmt.Sprintf("parent commit of commit %s/%s not found", err.Repo, err.ParentCommitID)
+}
+
+func (err ErrChildCommitNotFound) GRPCStatus() *status.Status {
+	return status.New(codes.NotFound, err.Error())
+}
+
+func IsErrCommitNotFound(err error) bool {
+	return errors.As(err, &ErrCommitNotFound{})
+}
+
+// ErrCommitMissingInfo is returned by CreateCommit() when a commitInfo is missing a field.
+type ErrCommitMissingInfo struct {
+	Field string
+}
+
+func (err ErrCommitMissingInfo) Error() string {
+	return fmt.Sprintf("commitInfo.%s is missing/nil", err.Field)
+}
+
+func (err ErrCommitMissingInfo) GRPCStatus() *status.Status {
+	return status.New(codes.FailedPrecondition, err.Error())
+}
+
+// ErrCommitAlreadyExists is returned by CreateCommit() when a commit with the same name already exists in postgres.
+type ErrCommitAlreadyExists struct {
+	CommitID string
+}
+
+// Error satisfies the error interface.
+func (err ErrCommitAlreadyExists) Error() string {
+	if id := err.CommitID; id != "" {
+		return fmt.Sprintf("commit %s already exists", id)
+	}
+	return "commit already exists"
+}
+
+func (err ErrCommitAlreadyExists) GRPCStatus() *status.Status {
+	return status.New(codes.AlreadyExists, err.Error())
+}
+
+type commitRow struct {
+	RowID          CommitID       `db:"int_id"`
+	CommitSetID    string         `db:"commit_set_id"`
+	CommitID       string         `db:"commit_id"`
+	RepoID         RepoID         `db:"repo_id"`
+	BranchID       sql.NullInt64  `db:"branch_id"`
+	BranchName     sql.NullString `db:"branch_name"`
+	Origin         string         `db:"origin"`
+	Description    string         `db:"description"`
+	StartTime      time.Time      `db:"start_time"`
+	FinishingTime  time.Time      `db:"finishing_time"`
+	FinishedTime   time.Time      `db:"finished_time"`
+	CompactingTime int64          `db:"compacting_time_s"`
+	ValidatingTime int64          `db:"validating_time_s"`
+	Error          string         `db:"error"`
+	Size           int64          `db:"size"`
+	RepoName       string         `db:"repo_name"`
+	RepoType       string         `db:"repo_type"`
+	ProjectName    string         `db:"proj_name"`
+	ParentCommit   *pfs.Commit    `db:"parent_commit"`
+	ChildCommits   []*pfs.Commit  `db:"child_commits"`
+}
+
+type AncestryOpt struct {
+	SkipChildren bool
+	SkipParent   bool
+}
+
+// CreateCommit creates an entry in the pfs.commits table.
+func CreateCommit(ctx context.Context, tx *pachsql.Tx, commitInfo *pfs.CommitInfo, opts ...AncestryOpt) error {
```

Comment:
A a good pattern is to return the id of the newly created commit, so we can re-use the id in downstream functions.